packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -749,3 +754,50 @@ export function reactHasJsxRuntime(): boolean {
 
   return false
 }
+
+export function ensureRequireEslintRules(config: Configuration): Configuration {
+  if (!config.module) {
+    config.module = {
+      rules: [],
+    }
+  }
+  // for fast refresh we want to ensure that that there is eslint rule running
+  // because user might have added their own `eslint-loader` let's check if there is one
+  // and adjust it to add the rule or append new loader with required rule
+  const rule = config.module?.rules.find(rule => {
```

Comment:
`config.module` is guaranteed to exist

```suggestion
  const rule = config.module.rules.find(rule => {
```